hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/KeyOutputStream.java

```diff
@@ -83,13 +85,13 @@ enum StreamAction {
   public static final Logger LOG =
       LoggerFactory.getLogger(KeyOutputStream.class);
 
-  private boolean closed;
+  private volatile boolean closed;
   private final Map<Class<? extends Throwable>, RetryPolicy> retryPolicyMap;
   private int retryCount;
   // how much of data is actually written yet to underlying stream
-  private long offset;
+  private volatile long offset;
```

Comment:
volatile boolean is fine.
But to do increment, AtomicLong is thread safe while volatile long isn't.